editor/editor_node.cpp

```diff
@@ -2381,12 +2381,24 @@ void EditorNode::_edit_current(bool p_skip_foreign) {
 
 	if (!inspector_only) {
 		EditorPlugin *main_plugin = editor_data.get_editor(current_obj);
+		Resource *current_res = Object::cast_to<Resource>(current_obj);
 
 		int plugin_index = 0;
 		for (; plugin_index < editor_table.size(); plugin_index++) {
 			if (editor_table[plugin_index] == main_plugin) {
+				// If button is not visible, then no plugin is active.
 				if (!main_editor_buttons[plugin_index]->is_visible()) {
-					main_plugin = nullptr; // If button is not visible, then no plugin is active.
+					Ref<EditorFeatureProfile> profile = feature_profile_manager->get_current_profile();
+					if (profile->is_feature_disabled(EditorFeatureProfile::FEATURE_EDITOR_SCRIPT) &&
+							!profile->is_feature_disabled(EditorFeatureProfile::FEATURE_SCRIPT)) {
+						if (main_plugin->get_name() == "Script" && current_res && !current_res->is_built_in() && (bool(EDITOR_GET("text_editor/external/use_external_editor")) || overrides_external_editor(current_obj))) {
```

Comment:
There is 4 nested conditions here, maybe it could be simplified.
The first `if` (with `== main_plugin`) could be changed to early continue.